### PR TITLE
Fix #43 - tabs for lists

### LINKED ISSUE #43: fix tabbing normal lists
currently, it incorrect converts them into headings

### PR DESCRIPTION
**Pull request by caffodian**
Needed to:

- make the tests do proper key events
- check that the selection is actually in a structured heading before attempting to increase heading level (this was causing h2s in lists before)

### CODE REVIEW

File: structuredheadings/plugin.js [hunk 1]
```diff
@@ -369,6 +369,13 @@
         if (editor.mode !== "wysiwyg") {
           return;
         }
+        var block = CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor);
+
+        // if the current block is not configured as a structured-heading-able block
+
+        if (block && editor.config.numberedElements.indexOf(block.getName()) === -1) {
+          return;
+        }
```
Comment (caffodian) [inline structuredheadings/plugin.js:378]: the actual fix is to just exit the listener if we're not in a relevant block

File: tests/structuredheadings/keyboardshortcuts.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
+/* eslint-disable */
 /* bender-tags: editor,unit */
-/* bender-ckeditor-plugins: wysiwygarea,structuredheadings,toolbar,basicstyles,dialog,richcombo */
+/* bender-ckeditor-plugins: wysiwygarea,structuredheadings,toolbar,basicstyles,dialog,richcombo,indentlist,list */
+/* eslint-enable */
```
Comment (caffodian) [inline tests/structuredheadings/keyboardshortcuts.js:4]: the line above (needed for bender config) is too damn long

File: tests/structuredheadings/keyboardshortcuts.js [hunk 3]
```diff
@@ -17,7 +19,10 @@
 
     "H2 to H3 from Tab": function () {
       this.editorBot.setHtmlWithSelection("<h2>^Heading</h2>");
-      this.editor.fire("key", { keyCode: tabKey });
+      this.editor.editable().fire(
+        "keydown",
+        new CKEDITOR.dom.event({ keyCode: tabKey }) // eslint-disable-line new-cap
+      );
```
Comment (caffodian) [inline tests/structuredheadings/keyboardshortcuts.js:25]: before, we weren't using actual events, just objects with the correct keys in place.  This fails once you bubble events up to the original listeners, who use more fields from the event objects

**Comment by bocribbz** [comment]
awesome! 👍

**Review by bocribbz — APPROVED** [review]

**Pull request merged**